courseraresearchexports/containers/client.py

```diff
@@ -0,0 +1,215 @@
+                raise RuntimeError('Container initialization failed.')
+
+            time.sleep(10)
+
+        logging.info('Initialized container {}.'.format(container_name_or_id))
+
+    except:
+        logging.error(
+            """Container initialization failed, check log for errors:\n{}"""
+            .format(docker_client.logs(container_name_or_id, tail=20)))
+        logging.error(
+            """If error persists, consider restarting your docker engine.""")
+        raise
+
+
+def create_from_folder(export_data_folder, docker_client,
+                       container_name='coursera-exports',
+                       database_name='coursera-exports'):
+    """
+    Using a folder containing a Coursera research export, create a docker
+     container with the export data loaded into a data base and start the
+     container
+    :param export_data_folder: folder where export data/scripts is stored
+    :param docker_client:
+    :param container_name:
+    :param database_name:
+    :return container_id:
+    """
+    if not docker_client.images(name=POSTGRES_DOCKER_IMAGE):
+        logging.info('Downloading image: {}'.format(POSTGRES_DOCKER_IMAGE))
+        docker_client.import_image(image=POSTGRES_DOCKER_IMAGE)
+
+    for existing_container in docker_client.containers(
+            all=True, filters={'name': container_name}):
+        logging.info('Removing existing container with name: {}'.format(
+            container_name))
+        docker_client.stop(existing_container)
+        docker_client.remove_container(existing_container)
+
+    logging.debug('Creating containers from {folder}'.format(
+        folder=export_data_folder))
+    container = docker_client.create_container(
+        image=POSTGRES_DOCKER_IMAGE,
+        name=container_name,
+        labels={COURSERA_DOCKER_LABEL: None, 'database_name': database_name},
+        volumes=['/mnt/exportData'],
+        host_config=docker_client.create_host_config(
+            binds=['{}:/mnt/exportData:ro'.format(export_data_folder)],
+            port_bindings={
+                5432: utils.get_next_available_port(list_all(
+                    docker_client))
+            }))
+
+    container_id = container['Id']
+
+    # copy containers initialization script to entrypoint
+    database_setup_script = """
+        createdb -U {user} {db}
+        cd /mnt/exportData
+        psql -e -U {user} -d {db} -f setup.sql
+        psql -e -U {user} -d {db} -f load.sql
+    """.format(user='postgres', db=database_name)
+
+    docker_client.put_archive(
+        container_id,  # using a named argument causes NullResource error
+        path='/docker-entrypoint-initdb.d/',
+        data=utils.create_tar_archive(
+            database_setup_script, name='init-user-db.sh'))
+
+    logging.info('Created container with id: {}'.format(container_id))
+
+    initialize(container_id, docker_client)
+
+    return container_id
+
+
+def create_from_export_request_id(export_request_id, docker_client,
```

Comment:
for db name - sure, container name, ideally it can be some id prefix followed by the inferred course slug/partner short name, that sounds fine.